### PR TITLE
add support for version 2

### PR DESCRIPTION
**Pull request by smoqadam**

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -21,7 +21,7 @@
         "elife/api": "^2.0",
         "elife/api-client": "dev-master",
         "elife/api-problem": "^1.1",
-        "elife/api-sdk": "dev-master",
```
Comment (nlisgo) [inline composer.json:24]: We can still rely upon `dev-master` in order to get this through

File: tests/src/Search/Web/TypeTest.php [hunk 2]
```diff
@@ -25,7 +25,7 @@ public function typeProvider() : Traversable
             'application/vnd.elife.search+json' => 200,
             'application/vnd.elife.search+json; version=0' => 406,
             'application/vnd.elife.search+json; version=1' => 200,
-            'application/vnd.elife.search+json; version=2' => 406,
+            'application/vnd.elife.search+json; version=2' => 200,
```
Comment (nlisgo) [inline tests/src/Search/Web/TypeTest.php:28]: We should add a test for version 3 to return 406.

**Review by nlisgo — CHANGES_REQUESTED** [review]
We also need a test to verify that the response header contains the appropriate Content-Type  value.

File: tests/src/Search/Web/TypeTest.php (hunk 2 above)
Comment (smoqadam) [reply tests/src/Search/Web/TypeTest.php:28]: @nlisgo added.

File: .docker/goaws.yaml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+Dev:                                # Another environment
+  Host: localhost
+  Port: 4100
+  # SqsPort: 9324
+  # SnsPort: 9292
+  Region: us-east-1
+  LogToFile: true
+  LogFile: ./goaws_messages.log
+  Queues:
+    - Name: search--dev
```
Comment (nlisgo) [inline .docker/goaws.yaml:10]: Doesn't belong in this PR.

File: docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,70 @@
+version: '3'
```
Comment (nlisgo) [inline docker-compose.yml:1]: Doesn't belong in this PR.

File: src/Search/Kernel.php [hunk 5]
```diff
@@ -399,10 +399,13 @@ public function dependencies(Application $app)
                 $config['endpoint'] = $app['config']['aws']['endpoint'];
             }
             if (!isset($app['config']['aws']['credential_file']) || false === $app['config']['aws']['credential_file']) {
-                $config['credentials'] = [
-                    'key' => $app['config']['aws']['key'],
-                    'secret' => $app['config']['aws']['secret'],
-                ];
+                $config['credentials'] = false;
```
Comment (nlisgo) [inline src/Search/Kernel.php:402]: Doesn't belong in this PR.

File: tests/src/Search/Web/TypeTest.php [hunk 6]
```diff
@@ -10,27 +11,56 @@ final class TypeTest extends ElasticTestCase
...
         foreach ($types as $type => $statusCode) {
             yield $type => [$type, $statusCode];
         }
     }
+
+    /**
+     * @test
+     * @dataProvider contentTypeProvider
+     */
+    public function testContentType($type, $contentType)
```
Comment (nlisgo) [inline tests/src/Search/Web/TypeTest.php:40]: add `string` type hints

File: tests/src/Search/Web/TypeTest.php [hunk 7]
```diff
@@ -10,27 +11,56 @@ final class TypeTest extends ElasticTestCase
...
+    {
+        $response = $this->performRequest($type);
+        $this->assertSame($contentType, $response->headers->get('Content-Type'));
+    }
+
+    public function contentTypeProvider()
```
Comment (nlisgo) [inline tests/src/Search/Web/TypeTest.php:46]: Should be `public function contentTypeProvider() : Traversable`

**Review by nlisgo — CHANGES_REQUESTED** [review]

**Comment by nlisgo** [comment]
I believe the error is being caused by the search response not being valid as reviewed-preprint is not present in the types array.

**Comment by nlisgo** [comment]
Tests are green but the new `reviewed-preprint` types value will be introduce regardless of the version requested. This isn't ideal but I'm happy to merge in as long as we confirm that journal search page doesn't react in a strange way. Let's talk about this tomorrow.

**Review by nlisgo — APPROVED** [review]
